src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -178,28 +184,25 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
     this.subscriptions.push(
       this.displayMode.view.subscribe(viewChange => {
         // Listen for view changes and move cells around depending on the current displayType
-        // remove cell views from display view
-        for (let i = this._scrollableCells.length; i > 0; i--) {
-          this._scrollableCells.detach();
-        }
-        // remove cell views from calculated view
-        for (let i = this._calculatedCells.length; i > 0; i--) {
-          this._calculatedCells.detach();
-        }
+        // detach cell views from display view
+        this.viewManager.detachAllViews(this._scrollableCells);
+        // detach cell views from calculated view
+        this.viewManager.detachAllViews(this._calculatedCells);
         if (viewChange === DatagridDisplayMode.CALCULATE) {
           this.displayCells = false;
-          this.dgCells.forEach(cell => {
-            this._calculatedCells.insert(cell._view);
-          });
+          this.viewManager.insertAllViews(this._calculatedCells, this.assignRawOrders(), true);
         } else {
           this.displayCells = true;
-          this.dgCells.forEach(cell => {
-            this._scrollableCells.insert(cell._view);
-          });
+          this.viewManager.insertAllViews(this._scrollableCells, this.assignRawOrders(), true);
         }
       }),
       this.expand.animate.subscribe(() => {
         this.expandAnimationTrigger = !this.expandAnimationTrigger;
+      }),
+      // A subscription that listens for view reordering
+      this.columnReorderService.reorderRequested.subscribe(reorderRequest => {
```

Comment:
Need to clean up subscriptions in several places.

Reply:
They get pushed into the array of subscriptions too :D  Does it look like it's not being cleaned up because it's in the same `.push()`? I could separate them if you prefer.

Reply:
Given the diff doesn't show the full code block, I didn't see the push above. Up to you if you want to make it more obvious.